src/test/unit/installer.test.ts

```diff
@@ -0,0 +1,49 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck ignore type checking in test files
+import { installTerraformLS } from '../../installer/installer';
+import { reporter } from '../../__mocks__/reporter';
+import { pathExists } from '../../installer/detector';
+import { window } from '../../__mocks__/vscode';
+
+jest.mock('../../installer/detector');
+describe('ls installer', () => {
+  describe('should install', () => {
+    let temp;
+
+    beforeEach(() => {
+      temp = require('temp').track();
+    });
+
+    test('when valid', async () => {
+      const installPath = temp.path('foo');
+
+      const release = {
+        getBuild: jest.fn((os: string, arch: string) => {
+          return {
+            url: 'https://releases.hashicorp.com/terraform-ls/0.24.0/terraform-ls_0.24.0_windows_amd64.zip',
+            filename: 'terraform-ls_0.24.0_windows_amd64.zip',
+          };
+        }),
+        download: jest.fn(),
+        verify: jest.fn(),
+        unpack: jest.fn(),
+      };
+
+      pathExists.mockImplementationOnce(() => {
+        return true;
+      });
+
+      window.withProgress.mockImplementationOnce(() => {
```

Comment:
There are a few things we need to do, to get this working.

First, since we're calling
```ts
vscode.workspace.fs.delete(vscode.Uri.file(zipfile))
```
inside the installer, I would add `delete: jest.fn(),` to our vscode workspace fs mock.

`withProgress` calls the passed `task` function with `vscode.Progress` and a cancellation token, so we need to replicate this in our test:
```ts
const report = jest.fn();
const token = {
  isCancellationRequested: false,
  onCancellationRequested: jest.fn(),
};
withProgress.mockImplementationOnce(async (_, task) => {
  task({ report }, token);
});
```

We can now use the expects, you already added and add even more:

```ts
expect(release.getBuild).toBeCalledTimes(1);
expect(withProgress).toBeCalledTimes(1);
expect(withProgress).toHaveBeenCalledWith(
  {
    cancellable: false,
    location: vscode.ProgressLocation.Window,
    title: 'Installing terraform-ls',
  },
  expect.any(Function),
);

expect(release.download).toBeCalledTimes(1);
expect(release.verify).toBeCalledTimes(1);
expect(release.unpack).toBeCalledTimes(1);
expect(vscode.workspace.fs.delete).toBeCalledTimes(1);
expect(report).toHaveBeenCalledTimes(4);
```
We could even test the arguments to all calls in greater detail, if we want.